pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -663,6 +684,8 @@ func (m *kubeGenericRuntimeManager) killContainersWithSyncResult(pod *v1.Pod, ru
 	wg.Wait()
 
 	if utilfeature.DefaultFeatureGate.Enabled(features.SidecarLifecycle) && len(sidecars) != 0 {
+		gracePeriodOverride = gracePeriodRemaining(pod, gracePeriodOverride, timeContainerKillStarted)
```

Comment:
Have we reached the agreement on sharing grace period? if not I'd still prefer that we go conservative on this. Or either sidecar or non-sidecar will get non-deterministic graceful period and can make developers confused and not feeling safe.

cc @mattjmcnaughton @tedyu @tomwans

Reply:
It seems sidecar containers don't have to respect the full grace period - since normal containers are gone.
A fraction of the grace period should suffice.

Reply:
yes we agreed last year with sig-node that grace period will be shared, this was reflected in an update to the KEP